url7f43a4752f_shalynnho_pokemonfactory_tgz-pJ8-Networking_ServerJ8/src/DeviceGraphicsDisplay/PartsRobotDisplay.java

```diff
@@ -0,0 +1,385 @@
+package DeviceGraphicsDisplay;
+
+import java.awt.Graphics2D;
+import java.awt.Image;
+import java.awt.Toolkit;
+import java.util.ArrayList;
+
+import javax.swing.JComponent;
+
+import Networking.Client;
+import Networking.Request;
+import Utils.Constants;
+import Utils.Location;
+import Utils.PartData;
+import factory.PartType;
+
+public class PartsRobotDisplay extends DeviceGraphicsDisplay {
+
+	private static Image partsRobotImage;
+	private static ArrayList<Image> armImage;
+
+	private final ArrayList<PartGraphicsDisplay> partArrayGraphics;
+
+	private Location loc, kitloc;
+	private final Location initialLocation;
+	private final Location currentLocation;
+	private final Location armLocation;
+
+	private PartType pt;
+	private int arm;
+	private int droparm;
+	private PartType ptdrop;
+	
+	private boolean home;
+	private boolean pickup, givekit;
+	private boolean gotpart;
+	private boolean gavepart;
+
+	private final ArrayList<Location> partStartLoc;
+	private final ArrayList<Location> armLoc;
+
+	public PartsRobotDisplay(Client prc) {
+		client = prc;
+
+		initialLocation = Constants.PARTS_ROBOT_LOC;
+		currentLocation = initialLocation;
+		armLocation = currentLocation;
+
+		partsRobotImage = Toolkit.getDefaultToolkit().getImage("src/images/parts_robot.png");
+		armImage = new ArrayList<Image>();
+		
+		for (int j = 0; j < 4; j++) {
+			armImage.add(Toolkit.getDefaultToolkit().getImage("src/images/parts_robot_arm.png"));
+		}
+
+		partArrayGraphics = new ArrayList<PartGraphicsDisplay>(4);
+		
+		
+
+		
+		home = true;
+		pickup = false;
+		givekit = false;
+
+		partStartLoc = new ArrayList<Location>();
+		partStartLoc.add(new Location(currentLocation.getX(), currentLocation.getY()));
+		partStartLoc.add(new Location(currentLocation.getX() + 30, currentLocation.getY()));
+		partStartLoc.add(new Location(currentLocation.getX(), currentLocation.getY() + 30));
+		partStartLoc.add(new Location(currentLocation.getX() + 30, currentLocation.getY() + 30));
+		
+		PartType pt = new PartType(null);
+		PartGraphicsDisplay pgd = new PartGraphicsDisplay(pt);
+		for (int i = 0; i<4;i++){
+		pgd.setLocation(partStartLoc.get(i));
+		partArrayGraphics.add(pgd);
+		}
+		
+		armLoc = new ArrayList<Location>();
+		armLoc.add(new Location(armLocation.getX(), armLocation.getY()));
+		armLoc.add(new Location(armLocation.getX(), armLocation.getY() + 30));
+		armLoc.add(new Location(armLocation.getX(), armLocation.getY() + 60));
+		armLoc.add(new Location(armLocation.getX(), armLocation.getY() + 90));
+		
+		gotpart = false;
+		gavepart = false;
+	}
+
+	@Override
+	public void draw(JComponent c, Graphics2D g) {
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

The cyclomatic complexity of this method is 45. By comparison, 99% of the methods in the CodeGuru reference dataset have a lower cyclomatic complexity. This indicates the method has a high number of decisions and it can make the logic difficult to understand and test. We recommend that you simplify this method or break it into multiple methods.